src/stage.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+CEPH_VERSION="$1"
+BASEOS_NAME="$2"
+BASEOS_TAG="$3"
+
+CP_CMD="cp --preserve=all --no-clobber"
+
+function stage_deepest_files () {
+  images_base_dir="${1%/}" # remove trailing '/'
+  image_dirs="$(find "${images_base_dir}/" -maxdepth 1 -mindepth 1 -type d -exec basename {} \;)"
+  for image_dir in ${image_dirs}; do
+    dest_dir="${STAGING_DIR}/${image_dir}"
+    mkdir --parents "${dest_dir}" # Create a corresponding image dir in staging
+    echo "Populating ${image_dir} staging from ${images_base_dir}/${image_dir}"
+    $CP_CMD --recursive "${images_base_dir}/${image_dir}"/. "${dest_dir}"/
+  done
+}
+
+function stage_shallower_files () {
+  images_base_dir="${1%/}" # remove trailing '/'
+  # Then for all images that we're staging, copy the files up to the image
+  # base but not within the image base
+  image_staging_dirs="$(find "${STAGING_DIR}/" -maxdepth 1 -mindepth 1 -type d)"
+  override_dir="${images_base_dir}" # start with deepest dir
+  while [ ! -z "${override_dir}" ]; do
+    echo "Populating all staging images from ${override_dir}"
+    find -D exec "${override_dir}/" -depth -maxdepth 1 -type f | \
+      while read file; do copy_to_multiple "${file}" "${image_staging_dirs}"; done
```

Comment:
This line triggers https://github.com/koalaman/shellcheck/wiki/SC2162.
Please fix or annotate to remove the warning.

Reply:
Next version will by Python linted with flake8 and line length 99.